### ISSUE TITLE
full series converter for `results.Null2NaN()`

### ISSUE THREAD

**Issue by DaveSkender** [DaveSkender/Stock.Indicators#981]
Add a public utility so you can simply do: `results.Null2NaN()` to make it easier to convert whole sets.
This will require unique indicator utilities if needed for anything more than `IResulableResult.Value` on base classes but can be done with a universal method for `tuple` results.

_Originally inspired by @mihakralj in https://github.com/DaveSkender/Stock.Indicators/issues/980#issuecomment-1362178492_

**Comment by DaveSkender** [comment]
I just noticed we already have a utility to do the `NaN` conversion on a whole results series.  This update would essentially add the `nullTo` parameters to the non-tuple `.ToCollection` families of converters.

```csharp
Collection<(DateTime Date, double? Value)> results = quotes
    .GetEma(..)
    .ToTupleCollection(NullTo.Null2NaN);
```

**Comment by DaveSkender** [comment]
After taking a look at this one, to do it properly would be too complex to handle for full classes (beyond this tuple model) given the default nullable class properties.  For now, I'm replacing the above use of the confusing `NullTo` enum with two simpler explicit utilities that work better.

```csharp
# indicator results converted to non-nullable NaN replacing nulls, retaining length
var results = quotes
    .GetEma(..)
    .ToTupleNaN();

# indicator results converted to tuple, retaining null values and length
var results = quotes
    .GetEma(..)
    .ToTupleCollection();

# a third option is also available for custom indicators:
# indicator results converted to non-nullable NaN with the removal of warmup null values
# which is suitable for preparing results to be chainable in custom indicators
var results = quotes
  .GetEma(..)
  .ToTupleChainable();
```

**Comment by DaveSkender** [comment]
This was implemented in v2.4.8
